rfcs/webtransport_h3_test_server.md

```diff
@@ -0,0 +1,94 @@
+
+`event` represents a WebTransport event such as a reception of stream data. Current proposed events can be found [here](https://bashi.github.io/wpt-wt-test-server-api/events.html).
+
+#### Alternative handler design
+
+Alternatively, we can define a set of functions which are called upon each event. This API pattern is similar to existing [H2 handlers](https://web-platform-tests.org/writing-tests/h2tests.html). The equivalent of the above example might look like:
+
+```python
+def datagram_received(session: WebTransportSession,
+                      data: bytes):
+    session.send_datagram(data=data)
+
+
+def bidirectional_stream_data_received(session: WebTransportSession,
+                                       stream_id: int,
+                                       data: bytes):
+    session.send_stream_data(stream_id_id=stream_id, data=data)
+```
+
+A major downside of this approach is that keeping API documents and implementations in sync would require extra works, as H2 handler API document duplicates comments from the source.
+
+Another downside of this approach is that we won't have editor/IDE supports on what kind of callback functions are available.
+
+TODO: Spend more time on the API design. People may prefer this alternative approach.
+
+### Handler lookup
+
+When the server receives an [extended CONNECT method](https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3-01#section-3.2) for WebTransport, it looks up a corresponding handler from the directory specified by a configuration. The configuration is passed by `wptserve`. If the server finds a corresponding handler the server creates a new WebTransport session with the handler and starts passing events to the handler.
+
+The server uses `":path"` header field of an extended CONNECT request to look up a handler. For example, if `":path"` is `/echo`, the server looks up `echo.webtransport.py` from the handler directory.
+
+### File name flags
+
+A new file name flag `.wt` is used to indicate a Python script is a WebTransport handler.
+
+`.h2` and `.h3` flags indicate that the underlying protocol (HTTP/2 or HTTP/3).
+
+### `wptserve` integration
+
+`wptserve` runs the WebTransport over HTTP/3 test server in a way similar to [pywebsocket](https://github.com/web-platform-tests/wpt/blob/246a32576020cb9c4241b7cfbc296f92d944ff6b/tools/serve/serve.py#L713): `wptserve` launches the server as a daemon and passes relevant configurations (port number, handler directory etc) to the server. The daemon is monitored by `wptserve` like other daemons.
+
+### Dependencies
+
+As of writing this RFC, the only dependency is `aioquic`.
+
+### `tools/quic` deprecation
+
+There are no ongoing standardlization efforts which require a general QUIC server. We might remove `tools/quic` directory at some point.
```

Comment:
I think I'd like to encourage this to happen sooner rather than later. The QUIC support is some additional complexity that we'd ideally do without.

Reply:
I see. We will remove them once we introduce the WebTransport over HTTP/3 test server.